pyo3-macros-backend/src/pymethod.rs

```diff
@@ -408,11 +406,18 @@ pub fn impl_arg_params(
         }
     }
 
+    let cls_name = if let Some(cls) = self_ {
+        quote! { Some(<#cls as pyo3::type_object::PyTypeInfo>::NAME) }
+    } else {
+        quote! { None }
+    };
+
     // create array of arguments, and then parse
     Ok(quote! {
         {
             const DESCRIPTION: pyo3::derive_utils::FunctionDescription = pyo3::derive_utils::FunctionDescription {
-                fname: _LOCATION,
+                cls_name: #cls_name,
+                func_name: _LOCATION,
```

Comment:
How about writing `#python_name` here, and then adding the `()` inside either `full_name` or each of the error messages?

This is the only use of `_LOCATION` and it's always seemed like awkward coupling to me - if we remove this usage then we can also delete `_LOCATION` from all of the `__wrap` functions.

Reply:
> How about writing #python_name here, and then adding the () inside either full_name() or each of the error messages?

I've also thought about it. But it gets a little complicated with getters & setters: https://github.com/PyO3/pyo3/blob/6ab61a15609e81de9cab1d8cbe4033c24a4638c4/pyo3-macros-backend/src/pymethod.rs#L261-L279

Reply:
I'm pretty sure `_LOCATION` is actually already unused in getters / setters - what happens if you just delete it there?

Reply:
Done in https://github.com/PyO3/pyo3/pull/1498/commits/e6374240b1b78c614aa6a2c72219336c67d016b1